osu.Game/Overlays/Changelog/ChangelogSupporterPromo.cs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Resources.Localisation.Web;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.Changelog
+{
+    public class ChangelogSupporterPromo : CompositeDrawable
+    {
+        private const float image_container_width = 164;
+
+        private readonly LinkFlowContainer supportLinkText;
+        private readonly TextFlowContainer supportNoteText;
+        private readonly Container imageContainer;
+
+        public ChangelogSupporterPromo()
+        {
+            RelativeSizeAxes = Axes.X;
+            AutoSizeAxes = Axes.Y;
+            Padding = new MarginPadding
+            {
+                Vertical = 20,
+                Horizontal = 50,
+            };
+
+            InternalChildren = new Drawable[]
+            {
+                new Container
+                {
+                    RelativeSizeAxes = Axes.X,
+                    AutoSizeAxes = Axes.Y,
+                    Masking = true,
+                    CornerRadius = 6,
+                    EdgeEffect = new EdgeEffectParameters
+                    {
+                        Type = EdgeEffectType.Shadow,
+                        Colour = Color4.Black.Opacity(0.25f),
+                        Offset = new Vector2(0, 1),
+                        Radius = 3,
+                    },
+                    Children = new Drawable[]
+                    {
+                        new Box
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Colour = Color4.Black.Opacity(0.3f),
```

Comment:
This box is much darker than it is web-side, and that's because in framework the shadow and the transparent black actually *stack*. As in, what CSS is doing when applying drop shadow to an element is *mask out the middle*, so that the middle can be filled in separately from the shadow with any transparency, while what framework seems to do is draw the shadow and then the 0.3 alpha box on top of it.

I'm not convinced that this is actually worth fixing, but just pointing it out.

Reply:
Ah, yeah I notice that. Will leave it as is for now.